<!-- HTML header for doxygen 1.8.14-->
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.17"/>
<meta name="viewport" content="width=device-width, initial-scale=1"/>
<meta name="description" content="iVar is a computational package that contains functions broadly useful for viral amplicon-based sequencing. Additional tools for metagenomic sequencing are actively being incorporated into iVar. While each of these functions can be accomplished using existing tools, iVar contains an intersection of functionality from multiple tools that are required to call iSNVs and consensus sequences from viral sequencing data across multiple replicates. We implemented the following functions in iVar: (1) trimming of primers and low-quality bases, (2) consensus calling, (3) variant calling - both iSNVs and insertions/deletions, and (4) identifying mismatches to primer sequences and excluding the corresponding reads from alignment files."/>
<meta name="keywords" content="variants calling,consensus,htslib,primal,primer,trimming,amplicon sequencing,ivar"/>
<meta name="author" content="Andersen Lab, The Scripps Research Institute"/>
<title>iVar: Installation</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/searchdata.js"></script>
<script type="text/javascript" src="search/search.js"></script>
<link href="customdoxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td id="projectalign" style="padding-left: 0.5em;">
   <div id="projectname">iVar
   </div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.17 -->
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
var searchBox = new SearchBox("searchBox", "search",false,'Search');
/* @license-end */
</script>
<script type="text/javascript" src="menudata.js"></script>
<script type="text/javascript" src="menu.js"></script>
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
$(function() {
  initMenu('',true,false,'search.php','Search');
  $(document).ready(function() { init_search(); });
});
/* @license-end */</script>
<div id="main-nav"></div>
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
</div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

</div><!-- top -->
<div class="PageDoc"><div class="header">
  <div class="headertitle">
<div class="title">Installation </div>  </div>
</div><!--header-->
<div class="contents">
<div class="toc"><h3>Table of Contents</h3>
<ul><li class="level1"><a href="#autotoc_md0">Installing via conda</a><ul><li class="level2"><a href="#autotoc_md1">Installing via homebrew</a></li>
</ul>
</li>
<li class="level1"><a href="#autotoc_md2">Installing on Mac</a><ul><li class="level2"><a href="#autotoc_md3">Installing build tools</a></li>
<li class="level2"><a href="#autotoc_md4">HTSlib installed using conda</a></li>
<li class="level2"><a href="#autotoc_md5">HTSlib installed from source</a></li>
<li class="level2"><a href="#autotoc_md6">Installing iVar</a></li>
</ul>
</li>
<li class="level1"><a href="#autotoc_md7">Installing on Linux</a><ul><li class="level2"><a href="#autotoc_md8">Installing build tools</a></li>
<li class="level2"><a href="#autotoc_md9">HTSlib installed using conda</a></li>
<li class="level2"><a href="#autotoc_md10">HTSlib installed from source</a></li>
<li class="level2"><a href="#autotoc_md11">Installing iVar</a></li>
</ul>
</li>
<li class="level1"><a href="#autotoc_md12">Running from Docker</a></li>
<li class="level1"><a href="#autotoc_md13">Contact</a></li>
</ul>
</div>
<div class="textblock"><h2>Dependencies</h2>
<ul>
<li><a href="http://www.htslib.org/download/">HTSlib</a></li>
<li><a href="https://gcc.gnu.org/">GCC</a> any version after v5.0. Support for c++11 standard required.</li>
</ul>
<p>Note:</p><ul>
<li>It is highly recommended that <a href="https://github.com/samtools/samtools">samtools</a> also be installed alongside iVar. iVar uses the output of samtools mpileup to call variants and generate consensus sequences. In addition, samtools <code>sort</code> and <code>index</code> commands are very useful to setup a pipeline using iVar.</li>
</ul>
<h1><a class="anchor" id="autotoc_md0"></a>
Installing via conda</h1>
<p>iVar is available on bioconda. To install conda, please use the <a href="https://conda.io/miniconda.html">miniconda</a> package. After intalling conda please add the following channels,</p>
<div class="fragment"><div class="line">conda config --add channels defaults</div>
<div class="line">conda config --add channels bioconda</div>
<div class="line">conda config --add channels conda-forge</div>
</div><!-- fragment --><p>To install iVar,</p>
<div class="fragment"><div class="line">conda install ivar</div>
</div><!-- fragment --><h2><a class="anchor" id="autotoc_md1"></a>
Installing via homebrew</h2>
<p>iVar can be installed using <a href="https://brew.sh/">Homebrew</a>.</p>
<div class="fragment"><div class="line">brew install brewsci/bio/ivar</div>
</div><!-- fragment --><h1><a class="anchor" id="autotoc_md2"></a>
Installing on Mac</h1>
<h2><a class="anchor" id="autotoc_md3"></a>
Installing build tools</h2>
<p><a href="https://developer.apple.com/xcode/">Xcode</a> from Apple is required to compile iVar (and other tools) from source. If you don't want to install the full Xcode package from the AppStore, you can install the Xcode command line tools,</p>
<div class="fragment"><div class="line">xcode-select --install</div>
</div><!-- fragment --><p><a href="https://www.gnu.org/software/automake/manual/html_node/Autotools-Introduction.html#Autotools-Introduction">GNU Autotools</a> is required to compile iVar from source.</p>
<p>To install Autotools using <a href="https://brew.sh/">homebrew</a> please use the command below,</p>
<div class="fragment"><div class="line">brew install autoconf automake libtool</div>
</div><!-- fragment --><h2><a class="anchor" id="autotoc_md4"></a>
HTSlib installed using conda</h2>
<p>HTSlib can be installed with <a href="https://conda.io/docs/">conda</a> using the command,</p>
<div class="fragment"><div class="line">conda install -c bioconda htslib</div>
</div><!-- fragment --><p>The conda binary is by default installed at /opt/. You can check the installation location by running the following command,</p>
<div class="fragment"><div class="line">which conda</div>
</div><!-- fragment --><p>The output of the command will be in this format - /opt/conda/bin/conda or /opt/anaconda2/bin/conda or /opt/anaconda3/bin/conda depending on whether you installed miniconda or anaconda.</p>
<p>If the output is for example, /opt/conda/bin/conda, then you can add the path to the lib folder to $LD_LIBRARY_PATH using the command below. You can add this to your ~/.bash_profile or ~/.bashrc to avoid rerunning the command everytime a new bash session starts.</p>
<div class="fragment"><div class="line">export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/opt/conda/lib</div>
</div><!-- fragment --><h2><a class="anchor" id="autotoc_md5"></a>
HTSlib installed from source</h2>
<p>Installation instructions and downloads for HTSlib can be found at <a href="http://www.htslib.org/download/">http://www.htslib.org/download/</a>.</p>
<p>If HTSlib is installed in a non standard location, please add the following to your .bash_profile so that iVar can find HTSlib dynamic libraries during runtime.</p>
<div class="fragment"><div class="line">export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/path/to/hts/lib/folder</div>
</div><!-- fragment --><h2><a class="anchor" id="autotoc_md6"></a>
Installing iVar</h2>
<p>To install iVar, run the following commands.</p>
<div class="fragment"><div class="line">./autogen.sh</div>
<div class="line">./configure</div>
<div class="line">make</div>
<div class="line">make install</div>
</div><!-- fragment --><p>If HTSlib was installed using conda, please run the following commands by supplying the prefix to the bin folder of the conda binary.</p>
<p>The prefix to the bin folder can be found using the command <code>which conda</code>. The output of the command will be in this format - /opt/conda/bin/conda or /opt/anaconda2/bin/conda or /opt/anaconda3/bin/conda depending on whether you installed miniconda or anaconda. For example, if the output of the command is /opt/conda/bin/conda, the prefix to the htslib bin folder will be /opt/conda. This can be supplied to ./configure &ndash;with-hts=/opt/conda.</p>
<div class="fragment"><div class="line">./autogen.sh</div>
<div class="line">./configure --with-hts=/prefix/to/bin/folder/with/HTSlib</div>
<div class="line">make</div>
<div class="line">make install</div>
</div><!-- fragment --><p>If HTSlib was installed in a non standard location, please run the following commands,</p>
<div class="fragment"><div class="line">./autogen.sh</div>
<div class="line">./configure --with-hts=/prefix/to/bin/folder/with/HTSlib</div>
<div class="line">make</div>
<div class="line">make install</div>
</div><!-- fragment --><p>To test installation just run, <code>ivar version</code> and you should get the following output,</p>
<div class="fragment"><div class="line">iVar version 1.0</div>
<div class="line"> </div>
<div class="line">Please raise issues and bug reports at https://github.com/andersen-lab/ivar/</div>
</div><!-- fragment --><h1><a class="anchor" id="autotoc_md7"></a>
Installing on Linux</h1>
<h2><a class="anchor" id="autotoc_md8"></a>
Installing build tools</h2>
<p><a href="https://www.gnu.org/software/automake/manual/html_node/Autotools-Introduction.html#Autotools-Introduction">GNU Autotools</a> is required to compile iVar from source.</p>
<p>To install Autotools using <a href="https://help.ubuntu.com/lts/serverguide/apt.html">APT</a> please use the command below,</p>
<div class="fragment"><div class="line">apt-get install autotools-dev</div>
</div><!-- fragment --><h2><a class="anchor" id="autotoc_md9"></a>
HTSlib installed using conda</h2>
<p>HTSlib can be installed with <a href="https://conda.io/docs/">conda</a> using the command,</p>
<div class="fragment"><div class="line">conda install -c bioconda htslib</div>
</div><!-- fragment --><p>The conda binary is by default installed at /opt/. You can check the installation location by running the following command,</p>
<div class="fragment"><div class="line">which conda</div>
</div><!-- fragment --><p>The output of the command will be in this format - /opt/conda/bin/conda or /opt/anaconda2/bin/conda or /opt/anaconda3/bin/conda depending on whether you installed miniconda or anaconda.</p>
<p>If the output is for example, /opt/conda/bin/conda, then you can add the path to the lib folder to $LD_LIBRARY_PATH using the command below. You can add this to your ~/.bash_profile or ~/.bashrc to avoid rerunning the command everytime a new bash session starts.</p>
<div class="fragment"><div class="line">export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/opt/conda/lib</div>
</div><!-- fragment --><h2><a class="anchor" id="autotoc_md10"></a>
HTSlib installed from source</h2>
<p>Installation instructions and downloads for HTSlib can be found at <a href="http://www.htslib.org/download/">http://www.htslib.org/download/</a>.</p>
<p>If HTSlib is installed in a non standard location, please add the following to your .bash_profile so that iVar can find HTSlib dynamic libraries during runtime.</p>
<div class="fragment"><div class="line">export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/path/to/hts/lib/folder</div>
</div><!-- fragment --><h2><a class="anchor" id="autotoc_md11"></a>
Installing iVar</h2>
<p>To install iVar, run the following commands.</p>
<div class="fragment"><div class="line">./autogen.sh</div>
<div class="line">./configure</div>
<div class="line">make</div>
<div class="line">make install</div>
</div><!-- fragment --><p>If HTSlib was installed using conda, please run the following commands by supplying the prefix to the bin folder of the conda binary.</p>
<p>The prefix to the bin folder can be found using the command <code>which conda</code>. The output of the command will be in this format - /opt/conda/bin/conda or /opt/anaconda2/bin/conda or /opt/anaconda3/bin/conda depending on whether you installed miniconda or anaconda. For example, if the output of the command is /opt/conda/bin/conda, the prefix to the htslib bin folder will be /opt/conda. This can be supplied to ./configure &ndash;with-hts=/opt/conda.</p>
<div class="fragment"><div class="line">./autogen.sh</div>
<div class="line">./configure --with-hts=/prefix/to/bin/folder/with/HTSlib</div>
<div class="line">make</div>
<div class="line">make install</div>
</div><!-- fragment --><p>If HTSlib was installed in a non standard location, please run the following commands,</p>
<div class="fragment"><div class="line">./autogen.sh</div>
<div class="line">./configure --with-hts=/prefix/to/bin/folder/with/HTSlib</div>
<div class="line">make</div>
<div class="line">make install</div>
</div><!-- fragment --><p>To test installation just run, <code>ivar version</code> and you should get the following output,</p>
<div class="fragment"><div class="line">iVar version 1.0</div>
<div class="line"> </div>
<div class="line">Please raise issues and bug reports at https://github.com/andersen-lab/ivar/</div>
</div><!-- fragment --><h1><a class="anchor" id="autotoc_md12"></a>
Running from Docker</h1>
<p>iVar can also be run via <a href="https://www.docker.com/">Docker</a>. Pull the docker image from <a href="https://hub.docker.com/">Docker Hub</a> using the following command,</p>
<div class="fragment"><div class="line">docker pull andersenlabapps/ivar</div>
</div><!-- fragment --><p>This docker image contains all the required dependencies to run iVar and the <a class="el" href="cookbookpage.html">pipelines</a> developed using iVar. You will have to attach a docker volume to get data into the docker container. Instructions to do so are in the <a href="https://docs.docker.com/storage/volumes/">Docker docs</a>.</p>
<p><a href="https://hub.docker.com/r/andersenlabapps/ivar/">iVar on Docker Hub</a></p>
<h1><a class="anchor" id="autotoc_md13"></a>
Contact</h1>
<p>For bug reports please email gkarthik[at]scripps.edu or raise an issue on Github. </p>
</div></div><!-- contents -->
</div><!-- PageDoc -->
<!-- HTML footer for doxygen 1.8.14-->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated by &#160;<a href="http://www.doxygen.org/index.html">
<img class="footer" src="doxygen.png" alt="doxygen"/>
</a> 1.8.17
</small></address>
</body>
</html>
