Two pipelines to call i\+S\+N\+Vs from known and unknown reference sequences have been written in \href{https://snakemake.readthedocs.io/en/stable/}{\texttt{ snakemake}}.

The two pipelines are distributed along with i\+Var and are present in the\href{https://github.com/andersen-lab/ivar/tree/master/pipeline}{\texttt{ pipeline/}} and \href{https://github.com/andersen-lab/ivar/tree/master/pipeline_field}{\texttt{ pipeline\+\_\+field/}} fodlers respectively.

For both pipelines, there are four parameters that will have to be set in beginning of the Snakefile.


\begin{DoxyCode}{0}
\DoxyCodeLine{in\_dir = "<input fastq files>"}
\DoxyCodeLine{out\_dir = "<output directory>"}
\DoxyCodeLine{bed = "<bed-\/file-\/with-\/primer-\/positions>"}
\DoxyCodeLine{ref="<path to reference fasta>"}
\end{DoxyCode}




\doxysubsection*{Call i\+S\+N\+Vs from samples with known reference sequence}

\href{https://github.com/andersen-lab/paper_2018_primalseq-ivar/blob/master/cookbook/CookBook.ipynb}{\texttt{ Link to Jupyter Notebook}}

\href{https://github.com/andersen-lab/ivar/tree/master/pipeline}{\texttt{ Link to pipeline}}

\doxysubsubsection*{Dependencies\+:}


\begin{DoxyItemize}
\item \href{https://github.com/andersen-lab/ivar}{\texttt{ i\+Var}}
\item \href{https://htslib.org/}{\texttt{ samtools}}
\item \href{https://github.com/lh3/bwa}{\texttt{ bwa}}
\item \href{https://bedtools.readthedocs.io/en/latest/}{\texttt{ bedtools}}
\end{DoxyItemize}

\doxysubsection*{Call i\+S\+N\+Vs from samples with unknown reference sequence}

\href{https://github.com/andersen-lab/ivar/tree/master/pipeline_field/}{\texttt{ Link to pipeline}}

\doxysubsubsection*{Dependencies\+:}


\begin{DoxyItemize}
\item \href{https://github.com/andersen-lab/ivar}{\texttt{ i\+Var}}
\item \href{https://htslib.org/}{\texttt{ samtools}}
\item \href{https://github.com/lh3/bwa}{\texttt{ bwa}}
\item \href{https://bedtools.readthedocs.io/en/latest/}{\texttt{ bedtools}} 
\end{DoxyItemize}